projects/plugins/jetpack/functions.opengraph.php

```diff
@@ -352,81 +352,287 @@ function jetpack_og_get_image( $width = 200, $height = 200, $deprecated = null )
-	if ( empty( $image ) && function_exists( 'blavatar_domain' ) ) {
+	if ( function_exists( 'blavatar_domain' ) ) {
 		$blavatar_domain = blavatar_domain( site_url() );
 		if ( blavatar_exists( $blavatar_domain ) ) {
-			$image['src']    = blavatar_url( $blavatar_domain, 'img', $width, false, true );
-			$image['width']  = $width;
-			$image['height'] = $height;
+			return array(
+				'src'    => blavatar_url( $blavatar_domain, 'img', $width, false, true ),
+				'width'  => $width,
+				'height' => $height,
+				'type'   => 'blavatar',
+			);
 		}
 	}
 
 	// Second fall back, Site Logo.
-	if ( empty( $image ) && ( function_exists( 'jetpack_has_site_logo' ) && jetpack_has_site_logo() ) ) {
+	if (
+		function_exists( 'jetpack_has_site_logo' )
+		&& jetpack_has_site_logo()
+	) {
 		$image_id = jetpack_get_site_logo( 'id' );
 		$logo     = wp_get_attachment_image_src( $image_id, 'full' );
 		if (
 			isset( $logo[0] ) && isset( $logo[1] ) && isset( $logo[2] )
 			&& ( _jetpack_og_get_image_validate_size( $logo[1], $logo[2], $width, $height ) )
 		) {
-			$image['src']      = $logo[0];
-			$image['width']    = $logo[1];
-			$image['height']   = $logo[2];
-			$image['alt_text'] = Jetpack_PostImages::get_alt_text( $image_id );
+			return array(
+				'src'    => $logo[0],
+				'width'  => $logo[1],
+				'height' => $logo[2],
+				'type'   => 'site_logo',
+			);
 		}
 	}
 
 	// Third fall back, Core Site Icon, if valid in size.
-	if ( empty( $image ) && has_site_icon() ) {
+	if ( has_site_icon() ) {
 		$image_id = get_option( 'site_icon' );
 		$icon     = wp_get_attachment_image_src( $image_id, 'full' );
 		if (
 			isset( $icon[0] ) && isset( $icon[1] ) && isset( $icon[2] )
 			&& ( _jetpack_og_get_image_validate_size( $icon[1], $icon[2], $width, $height ) )
 		) {
-			$image['src']      = $icon[0];
-			$image['width']    = $icon[1];
-			$image['height']   = $icon[2];
-			$image['alt_text'] = Jetpack_PostImages::get_alt_text( $image_id );
+			return array(
+				'src'    => $icon[0],
+				'width'  => $icon[1],
+				'height' => $icon[2],
+				'type'   => 'site_icon',
+			);
 		}
 	}
 
-	// Final fall back, blank image.
-	if ( empty( $image ) ) {
-		/**
-		 * Filter the default Open Graph Image tag, used when no Image can be found in a post.
-		 *
-		 * @since 3.0.0
-		 *
-		 * @param string $str Default Image URL.
-		 */
-		$image['src'] = apply_filters( 'jetpack_open_graph_image_default', 'https://s0.wp.com/i/blank.jpg' );
+	return array(
+		'src'    => '',
+		'width'  => $width,
+		'height' => $height,
+		'type'   => 'blank',
+	);
+}
+
+/**
+ * Get the site's fallback image.
+ *
+ * @since $$next-version$$
+ *
+ * @return string
+ */
+function jetpack_og_get_site_fallback_blank_image() {
+	/**
+	 * Filter the default Open Graph Image tag, used when no Image can be found in a post.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param string $str Default Image URL.
+	 */
+	return apply_filters( 'jetpack_open_graph_image_default', 'https://s0.wp.com/i/blank.jpg' );
```

Comment:
It seems a bit odd to have an external image as the default fallback even for offline sites, no?

Reply:
That's a good point. It's nothing new though, it's been that way for the past 12 years, since 5ebbf7eaa0e09abef151affeb807441acc36de58. Since we haven't received any complaints in all those years, I'm tempted to leave it that way.